corehq/apps/hqwebapp/static/hqwebapp/js/inactivity.js

```diff
@@ -118,6 +124,17 @@ hqDefine('hqwebapp/js/inactivity', [
             $warningModal.find(".btn-primary").focus();
         });
 
+        // Keep track of when user is actively typing
+        $("body").on("keypress", _.throttle(function () {
+            keyboardOrMouseActive = true;
+        }, 100, {trailing: false}));
+        $("body").on("keypress", _.debounce(function () {
+            keyboardOrMouseActive = false;
+            if (warningActive) {
+                showWarningModal();
+            }
+        }, 500));
```

Comment:
I had not seen that SO answer. 1 sec is about how long it takes to move your hand from mouse to keyboard or vice versa, which is a much bigger gesture than moving between keyboard keys. So I'm estimating that 500ms is enough of a delay that you're probably not steadily typing. I did some testing (console logging when HQ thought I was or wasn't typing) and it felt natural enough, though of course typing speed varies. I don't want the delay to be so long that the user possibly doesn't see the warning at all before they get logged out, since that's much more disruptive than the warning. I'd be up for increasing it to 1 sec or maybe 2, but probably not more than that.